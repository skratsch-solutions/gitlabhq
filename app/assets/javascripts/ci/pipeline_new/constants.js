import { __ } from '~/locale';
import { DEFAULT_DEBOUNCE_AND_THROTTLE_MS } from '~/lib/utils/constants';

export const VARIABLE_TYPE = 'env_var';
export const FILE_TYPE = 'file';
export const DEBOUNCE_REFS_SEARCH_MS = DEFAULT_DEBOUNCE_AND_THROTTLE_MS;
export const CONFIG_VARIABLES_TIMEOUT = 5000;
export const BRANCH_REF_TYPE = 'branch';
export const TAG_REF_TYPE = 'tag';

export const IDENTITY_VERIFICATION_REQUIRED_ERROR = __(
  'Identity verification is required in order to run CI jobs',
);
