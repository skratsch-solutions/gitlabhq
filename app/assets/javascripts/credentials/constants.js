import { GlFilteredSearchToken } from '@gitlab/ui';
import { s__, __ } from '~/locale';
import {
  OPERATORS_AFTER_BEFORE,
  OPERATORS_IS,
} from '~/vue_shared/components/filtered_search_bar/constants';
import DateToken from '~/vue_shared/components/filtered_search_bar/tokens/date_token.vue';

export const TOKENS = [
  {
    icon: 'key',
    title: s__('CredentialsInventory|Type'),
    type: 'filter',
    token: GlFilteredSearchToken,
    operators: OPERATORS_IS,
    unique: true,
    options: [
      {
        value: 'personal_access_tokens',
        title: s__('CredentialsInventory|Personal access tokens'),
      },
      { value: 'ssh_keys', title: s__('CredentialsInventory|SSH keys') },
      {
        value: 'resource_access_tokens',
        title: s__('CredentialsInventory|Project and group access tokens'),
      },
      { value: 'gpg_keys', title: s__('CredentialsInventory|GPG keys') },
    ],
  },
  {
    icon: 'status',
    title: s__('CredentialsInventory|State'),
    type: 'state',
    token: GlFilteredSearchToken,
    operators: OPERATORS_IS,
    unique: true,
    options: [
      { value: 'active', title: s__('CredentialsInventory|Active') },
      { value: 'inactive', title: s__('CredentialsInventory|Inactive') },
    ],
  },
  {
    icon: 'remove',
    title: s__('CredentialsInventory|Revoked'),
    type: 'revoked',
    token: GlFilteredSearchToken,
    operators: OPERATORS_IS,
    unique: true,
    options: [{ value: 'true', title: __('Yes') }],
  },
  {
    icon: 'history',
    title: s__('CredentialsInventory|Created date'),
    type: 'created',
    token: DateToken,
    operators: OPERATORS_AFTER_BEFORE,
    unique: true,
  },
  {
    icon: 'history',
    title: s__('CredentialsInventory|Expiration date'),
    type: 'expires',
    token: DateToken,
    operators: OPERATORS_AFTER_BEFORE,
    unique: true,
  },
  {
    icon: 'history',
    title: s__('CredentialsInventory|Last used date'),
    type: 'last_used',
    token: DateToken,
    operators: OPERATORS_AFTER_BEFORE,
    unique: true,
  },
];
