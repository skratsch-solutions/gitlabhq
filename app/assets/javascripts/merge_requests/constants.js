export const MAX_LIST_SIZE = 10;
