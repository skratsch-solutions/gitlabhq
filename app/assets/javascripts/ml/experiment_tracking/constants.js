import { s__ } from '~/locale';
import { helpPagePath } from '~/helpers/help_page_helper';

export const FEATURE_NAME = s__('MlExperimentTracking|Machine learning experiment tracking');

export const FEATURE_FEEDBACK_ISSUE = 'https://gitlab.com/gitlab-org/gitlab/-/issues/381660';
export const ROUTE_DETAILS = 'details';
export const ROUTE_CANDIDATES = 'candidates';
export const ROUTE_PERFORMANCE = 'performance';
export const GRAPHQL_PAGE_SIZE = 30;
export const CANDIDATES_DOCS_PATH = helpPagePath(
  'user/project/ml/experiment_tracking/mlflow_client.md',
  { anchor: 'logging-runs-to-a-model' },
);
