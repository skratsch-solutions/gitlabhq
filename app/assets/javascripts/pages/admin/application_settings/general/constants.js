// used in EE
export const SEAT_CONTROL = {};
