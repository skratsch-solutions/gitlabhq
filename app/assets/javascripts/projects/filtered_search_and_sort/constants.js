import { __ } from '~/locale';

export const SORT_OPTION_NAME = {
  value: 'name',
  text: __('Name'),
};

export const SORT_OPTION_CREATED = {
  value: 'created',
  text: __('Created'),
};

export const SORT_OPTION_UPDATED = {
  value: 'latest_activity',
  text: __('Updated'),
};

export const SORT_OPTION_STARS = {
  value: 'stars',
  text: __('Stars'),
};

export const SORT_DIRECTION_ASC = 'asc';
export const SORT_DIRECTION_DESC = 'desc';

export const FILTERED_SEARCH_TERM_KEY = 'name';
export const FILTERED_SEARCH_NAMESPACE = 'explore';
export const SORT_OPTIONS = [
  SORT_OPTION_NAME,
  SORT_OPTION_CREATED,
  SORT_OPTION_UPDATED,
  SORT_OPTION_STARS,
];
