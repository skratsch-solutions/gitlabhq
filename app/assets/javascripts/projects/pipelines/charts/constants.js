export const CHART_CONTAINER_HEIGHT = 300;

export const INNER_CHART_HEIGHT = 200;

export const X_AXIS_LABEL_ROTATION = 45;

export const X_AXIS_TITLE_OFFSET = 60;

export const ONE_WEEK_AGO_DAYS = 7;
export const ONE_MONTH_AGO_DAYS = 31;
export const ONE_YEAR_AGO_DAYS = 365;

export const DATE_RANGE_LAST_WEEK = 7;
export const DATE_RANGE_LAST_30_DAYS = 30;
export const DATE_RANGE_LAST_90_DAYS = 90;
export const DATE_RANGE_LAST_180_DAYS = 180;

export const DEFAULT = 'default';
export const PARSE_FAILURE = 'parse_failure';
export const LOAD_ANALYTICS_FAILURE = 'load_analytics_failure';
export const LOAD_PIPELINES_FAILURE = 'load_analytics_failure';
export const UNSUPPORTED_DATA = 'unsupported_data';

export const SNOWPLOW_LABEL = 'redis_hll_counters.analytics.analytics_total_unique_counts_monthly';
export const SNOWPLOW_SCHEMA = 'iglu:com.gitlab/gitlab_service_ping/jsonschema/1-0-1';
export const SNOWPLOW_DATA_SOURCE = 'redis_hll';
