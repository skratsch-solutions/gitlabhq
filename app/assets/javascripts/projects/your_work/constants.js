import { __ } from '~/locale';
import projectsQuery from './graphql/queries/projects.query.graphql';
import userProjectsQuery from './graphql/queries/user_projects.query.graphql';

const transformSortToUpperCase = (variables) => ({
  ...variables,
  sort: variables.sort.toUpperCase(),
});

export const CONTRIBUTED_TAB = {
  text: __('Contributed'),
  value: 'contributed',
  query: userProjectsQuery,
  variables: { contributed: true },
  queryPath: 'currentUser.contributedProjects',
  transformVariables: transformSortToUpperCase,
};

export const STARRED_TAB = {
  text: __('Starred'),
  value: 'starred',
  query: userProjectsQuery,
  variables: { starred: true },
  queryPath: 'currentUser.starredProjects',
  transformVariables: transformSortToUpperCase,
};

export const PERSONAL_TAB = {
  text: __('Personal'),
  value: 'personal',
  query: projectsQuery,
  variables: { personal: true },
  queryPath: 'projects',
};

export const MEMBER_TAB = {
  text: __('Member'),
  value: 'member',
  query: projectsQuery,
  variables: { membership: true },
  queryPath: 'projects',
};

export const INACTIVE_TAB = {
  text: __('Inactive'),
  value: 'inactive',
  query: projectsQuery,
  variables: { archived: 'ONLY', membership: true },
  queryPath: 'projects',
};

export const PROJECT_DASHBOARD_TABS = [
  CONTRIBUTED_TAB,
  STARRED_TAB,
  PERSONAL_TAB,
  MEMBER_TAB,
  INACTIVE_TAB,
];

export const BASE_ROUTE = '/dashboard/projects';

export const ROOT_ROUTE_NAME = 'root';

export const DASHBOARD_ROUTE_NAME = 'dashboard';

export const PROJECTS_DASHBOARD_ROUTE_NAME = 'projects-dashboard';

export const CUSTOM_DASHBOARD_ROUTE_NAMES = [
  ROOT_ROUTE_NAME,
  DASHBOARD_ROUTE_NAME,
  PROJECTS_DASHBOARD_ROUTE_NAME,
];

export const FILTERED_SEARCH_NAMESPACE = 'dashboard';
export const FILTERED_SEARCH_TERM_KEY = 'name';
export const FILTERED_SEARCH_TOKEN_LANGUAGE = 'language';
export const FILTERED_SEARCH_TOKEN_MIN_ACCESS_LEVEL = 'min_access_level';
