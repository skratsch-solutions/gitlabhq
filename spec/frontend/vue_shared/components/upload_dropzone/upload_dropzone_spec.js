import { GlAnimatedUploadIcon, GlSprintf } from '@gitlab/ui';
import { nextTick } from 'vue';
import { shallowMountExtended } from 'helpers/vue_test_utils_helper';
import UploadDropzone from '~/vue_shared/components/upload_dropzone/upload_dropzone.vue';
import { VALID_DESIGN_FILE_MIMETYPE } from '~/work_items/components/design_management/constants';

describe('Upload dropzone component', () => {
  let wrapper;

  const mockDragEvent = ({ types = ['Files'], files = [], items = [] }) => {
    return { dataTransfer: { types, files, items } };
  };

  const findDropzoneCard = () => wrapper.find('.upload-dropzone-card');
  const findDropzoneArea = () => wrapper.findByTestId('dropzone-area');
  const findIcon = () => wrapper.findComponent(GlAnimatedUploadIcon);
  const findUploadText = () => wrapper.findByTestId('upload-text').text();
  const findFileInput = () => wrapper.find('input[type="file"]');

  function createComponent({ slots = {}, props = {} } = {}) {
    wrapper = shallowMountExtended(UploadDropzone, {
      slots,
      propsData: {
        displayAsCard: true,
        ...props,
      },
      stubs: {
        GlSprintf,
      },
    });
  }

  describe('when slot provided', () => {
    it('renders dropzone with slot content', () => {
      createComponent({
        slots: {
          default: ['<div>dropzone slot</div>'],
        },
      });

      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('when no slot provided', () => {
    it('renders default dropzone card', () => {
      createComponent();

      expect(wrapper.element).toMatchSnapshot();
    });

    it('triggers click event on file input element when clicked', () => {
      createComponent();
      const clickSpy = jest.spyOn(wrapper.find('input').element, 'click');

      findDropzoneCard().trigger('click');
      expect(clickSpy).toHaveBeenCalled();
    });
  });

  describe('upload text', () => {
    it.each`
      collection    | description                   | props                            | expected
      ${'multiple'} | ${'by default'}               | ${null}                          | ${'files to attach'}
      ${'singular'} | ${'when singleFileSelection'} | ${{ singleFileSelection: true }} | ${'file to attach'}
    `('displays $collection version $description', ({ props, expected }) => {
      createComponent({ props });

      expect(findUploadText()).toContain(expected);
    });
  });

  describe('when dragging', () => {
    it.each`
      description                  | eventPayload
      ${'is empty'}                | ${{}}
      ${'contains text'}           | ${mockDragEvent({ types: ['text'] })}
      ${'contains files and text'} | ${mockDragEvent({ types: ['Files', 'text'] })}
      ${'contains files'}          | ${mockDragEvent({ types: ['Files'] })}
    `('renders correct template when drag event $description', async ({ eventPayload }) => {
      createComponent();

      wrapper.trigger('dragenter', eventPayload);
      await nextTick();
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders correct template when dragging stops', async () => {
      createComponent();

      wrapper.trigger('dragenter');

      await nextTick();
      wrapper.trigger('dragleave');

      await nextTick();
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('when dragging with design upload overlay enabled', () => {
    const findDesignUploadOverlay = () => wrapper.findByTestId('design-upload-overlay');
    const triggerDragEvents = async (dragEvent) => {
      wrapper.trigger('dragenter', dragEvent);
      await nextTick();

      wrapper.trigger('dragover', dragEvent);
      await nextTick();
    };

    beforeEach(() => {
      createComponent({
        props: {
          showUploadDesignOverlay: true,
          validateDesignUploadOnDragover: true,
          uploadDesignOverlayText: 'Drop your images to start the upload.',
          acceptDesignFormats: VALID_DESIGN_FILE_MIMETYPE.mimetype,
        },
      });
    });

    it('renders component with requires classes when design upload overlay is true', async () => {
      const dragEvent = mockDragEvent({
        types: ['Files', 'image'],
        items: [{ type: 'image/png' }],
      });

      await triggerDragEvents(dragEvent);

      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders design upload overlay with text on drag of valid design', async () => {
      const dragEvent = mockDragEvent({
        types: ['Files', 'image'],
        items: [{ type: 'image/png' }],
      });

      await triggerDragEvents(dragEvent);

      const designUploadOverlay = findDesignUploadOverlay();
      expect(designUploadOverlay.exists()).toBe(true);
      expect(designUploadOverlay.isVisible()).toBe(true);
      expect(designUploadOverlay.findComponent(GlAnimatedUploadIcon).exists()).toBe(true);
      expect(designUploadOverlay.text()).toBe('Drop your images to start the upload.');
    });

    it('does not render design upload overlay on drag of invalid design', async () => {
      const dragEvent = mockDragEvent({
        types: ['Files', 'video'],
        items: [{ type: 'video/quicktime' }],
      });

      await triggerDragEvents(dragEvent);

      const designUploadOverlay = findDesignUploadOverlay();
      expect(designUploadOverlay.exists()).toBe(false);
    });
  });

  describe('when dropping', () => {
    it('emits upload event', async () => {
      createComponent();
      const mockFile = { name: 'test', type: 'image/jpg' };
      const mockEvent = mockDragEvent({ files: [mockFile] });

      wrapper.trigger('dragenter', mockEvent);

      await nextTick();
      wrapper.trigger('drop', mockEvent);

      await nextTick();
      expect(wrapper.emitted('change')).toEqual([[[mockFile]]]);
    });
  });

  describe('ondrop', () => {
    describe('when drag data is valid', () => {
      it('emits upload event for valid files', () => {
        createComponent();

        const mockFile = { type: 'image/jpg' };
        const mockEvent = mockDragEvent({ files: [mockFile] });

        wrapper.trigger('drop', mockEvent);
        expect(wrapper.emitted('change')).toEqual([[[mockFile]]]);
      });

      it('emits error event when files are invalid', () => {
        createComponent();
        const mockEvent = mockDragEvent({ files: [{ type: 'audio/midi' }] });

        wrapper.trigger('drop', mockEvent);
        expect(wrapper.emitted()).toHaveProperty('error');
      });

      it('allows validation function to be overwritten', () => {
        createComponent({ props: { isFileValid: () => true } });

        const mockEvent = mockDragEvent({ files: [{ type: 'audio/midi' }] });

        wrapper.trigger('drop', mockEvent);
        expect(wrapper.emitted()).not.toHaveProperty('error');
      });

      describe('singleFileSelection = true', () => {
        it('emits a single file on drop', () => {
          createComponent({
            props: { singleFileSelection: true },
          });

          const mockFile = { type: 'image/jpg' };
          const mockEvent = mockDragEvent({ files: [mockFile] });

          wrapper.trigger('drop', mockEvent);
          expect(wrapper.emitted('change')).toEqual([[mockFile]]);
        });
      });
    });
  });

  it('applies correct classes when displaying as a standalone item', () => {
    createComponent({ props: { displayAsCard: false } });
    expect(findDropzoneArea().classes()).not.toContain('gl-flex-col');
    expect(findIcon().classes()).toEqual(['gl-mr-3']);
  });

  it('applies correct classes when displaying in card mode', () => {
    createComponent({ props: { displayAsCard: true } });
    expect(findDropzoneArea().classes()).toContain('gl-flex-col');
    expect(findIcon().classes()).toEqual(['gl-mb-3']);
  });

  it('animates icon on hover', async () => {
    createComponent();

    findDropzoneCard().trigger('mouseenter');
    await nextTick();

    expect(findIcon().props('isOn')).toEqual(true);
  });

  it('does not animate icon on mouse leave', async () => {
    createComponent();

    findDropzoneCard().trigger('mouseleave');
    await nextTick();

    expect(findIcon().props('isOn')).toEqual(false);
  });

  it('correctly overrides description and drop messages', () => {
    createComponent({
      props: {
        dropToStartMessage: 'Test drop-to-start message.',
        validFileMimetypes: ['image/jpg', 'image/jpeg'],
      },
      slots: {
        'upload-text': '<span>Test %{linkStart}description%{linkEnd} message.</span>',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('correctly overrides single upload messages', () => {
    createComponent({
      props: {
        singleFileSelection: true,
        uploadSingleMessage: 'Drop or select file to attach',
      },
    });
    expect(findUploadText()).toContain('Drop or select file to attach');
  });

  it('correctly overrides multiple upload messages', () => {
    createComponent({
      props: {
        singleFileSelection: false,
        uploadMultipleMessage: 'Drop or select files to attach',
      },
    });

    expect(findUploadText()).toContain('Drop or select files to attach');
  });

  describe('file input form name', () => {
    it('applies inputFieldName as file input name', () => {
      createComponent({ props: { inputFieldName: 'test_field_name' } });
      expect(findFileInput().attributes('name')).toBe('test_field_name');
    });

    it('uses default file input name if no inputFieldName provided', () => {
      createComponent();
      expect(findFileInput().attributes('name')).toBe('upload_file');
    });
  });

  describe('updates file input files value', () => {
    // NOTE: the component assigns dropped files from the drop event to the
    // input.files property. There's a restriction that nothing but a FileList
    // can be assigned to this property. While FileList can't be created
    // manually: it has no constructor. And currently there's no good workaround
    // for jsdom. So we have to stub the file input in vm.$refs to ensure that
    // the files property is updated. This enforces following tests to know a
    // bit too much about the SUT internals See this thread for more details on
    // FileList in jsdom: https://github.com/jsdom/jsdom/issues/1272
    function stubFileInputOnWrapper() {
      const fakeFileInput = { files: [] };
      wrapper.vm.$refs.fileUpload = fakeFileInput;
    }

    it('assigns dragged files to the input files property', async () => {
      const mockFile = { name: 'test', type: 'image/jpg' };
      const mockEvent = mockDragEvent({ files: [mockFile] });
      createComponent({ props: { shouldUpdateInputOnFileDrop: true } });
      stubFileInputOnWrapper();

      wrapper.trigger('dragenter', mockEvent);
      await nextTick();
      wrapper.trigger('drop', mockEvent);
      await nextTick();

      expect(wrapper.vm.$refs.fileUpload.files).toEqual([mockFile]);
    });

    it('throws an error when multiple files are dropped on a single file input dropzone', async () => {
      const mockFile = { name: 'test', type: 'image/jpg' };
      const mockEvent = mockDragEvent({ files: [mockFile, mockFile] });
      createComponent({ props: { shouldUpdateInputOnFileDrop: true, singleFileSelection: true } });
      stubFileInputOnWrapper();

      wrapper.trigger('dragenter', mockEvent);
      await nextTick();
      wrapper.trigger('drop', mockEvent);
      await nextTick();

      expect(wrapper.vm.$refs.fileUpload.files).toEqual([]);
      expect(wrapper.emitted('error')).toHaveLength(1);
    });
  });
});
