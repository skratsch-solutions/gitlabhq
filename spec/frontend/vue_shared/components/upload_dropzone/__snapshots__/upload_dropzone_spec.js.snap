// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload dropzone component correctly overrides description and drop messages 1`] = `
<div
  class="gl-relative gl-w-full"
>
  <button
    class="card gl-h-full gl-items-center gl-justify-center gl-mb-0 gl-px-5 gl-py-4 gl-w-full upload-dropzone-border upload-dropzone-card"
    type="button"
  >
    <div
      class="gl-flex gl-flex-col gl-items-center gl-justify-center gl-text-center"
      data-testid="dropzone-area"
    >
      <gl-animated-upload-icon-stub
        class="gl-mb-3"
      />
      <p
        class="gl-mb-0"
        data-testid="upload-text"
      >
        <span>
          Test %{linkStart}description%{linkEnd} message.
        </span>
      </p>
    </div>
  </button>
  <input
    accept="image/jpg,image/jpeg"
    class="hide"
    multiple=""
    name="upload_file"
    type="file"
  />
  <transition-stub
    name="upload-dropzone-fade"
  >
    <div
      class="card gl-absolute gl-flex gl-h-full gl-items-center gl-justify-center gl-p-4 gl-w-full upload-dropzone-border upload-dropzone-overlay"
      style="display: none;"
    >
      <div
        class="gl-max-w-1/2 gl-text-center"
      >
        <h3>
          Incoming!
        </h3>
        <span>
          Test drop-to-start message.
        </span>
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`Upload dropzone component when dragging renders correct template when drag event contains files 1`] = `
<div
  class="gl-relative gl-w-full"
>
  <button
    class="card gl-h-full gl-items-center gl-justify-center gl-mb-0 gl-px-5 gl-py-4 gl-w-full upload-dropzone-border upload-dropzone-card"
    type="button"
  >
    <div
      class="gl-flex gl-flex-col gl-items-center gl-justify-center gl-text-center"
      data-testid="dropzone-area"
    >
      <gl-animated-upload-icon-stub
        class="gl-mb-3"
      />
      <p
        class="gl-mb-0"
        data-testid="upload-text"
      >
        Drop or
        <gl-link-stub>
          upload
        </gl-link-stub>
        files to attach
      </p>
    </div>
  </button>
  <input
    accept="image/*"
    class="hide"
    multiple=""
    name="upload_file"
    type="file"
  />
  <transition-stub
    name="upload-dropzone-fade"
  >
    <div
      class="card gl-absolute gl-flex gl-h-full gl-items-center gl-justify-center gl-p-4 gl-w-full upload-dropzone-border upload-dropzone-overlay"
      style=""
    >
      <div
        class="gl-max-w-1/2 gl-text-center"
      >
        <h3>
          Incoming!
        </h3>
        <span>
          Drop your files to start your upload.
        </span>
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`Upload dropzone component when dragging renders correct template when drag event contains files and text 1`] = `
<div
  class="gl-relative gl-w-full"
>
  <button
    class="card gl-h-full gl-items-center gl-justify-center gl-mb-0 gl-px-5 gl-py-4 gl-w-full upload-dropzone-border upload-dropzone-card"
    type="button"
  >
    <div
      class="gl-flex gl-flex-col gl-items-center gl-justify-center gl-text-center"
      data-testid="dropzone-area"
    >
      <gl-animated-upload-icon-stub
        class="gl-mb-3"
      />
      <p
        class="gl-mb-0"
        data-testid="upload-text"
      >
        Drop or
        <gl-link-stub>
          upload
        </gl-link-stub>
        files to attach
      </p>
    </div>
  </button>
  <input
    accept="image/*"
    class="hide"
    multiple=""
    name="upload_file"
    type="file"
  />
  <transition-stub
    name="upload-dropzone-fade"
  >
    <div
      class="card gl-absolute gl-flex gl-h-full gl-items-center gl-justify-center gl-p-4 gl-w-full upload-dropzone-border upload-dropzone-overlay"
      style=""
    >
      <div
        class="gl-max-w-1/2 gl-text-center"
      >
        <h3>
          Incoming!
        </h3>
        <span>
          Drop your files to start your upload.
        </span>
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`Upload dropzone component when dragging renders correct template when drag event contains text 1`] = `
<div
  class="gl-relative gl-w-full"
>
  <button
    class="card gl-h-full gl-items-center gl-justify-center gl-mb-0 gl-px-5 gl-py-4 gl-w-full upload-dropzone-border upload-dropzone-card"
    type="button"
  >
    <div
      class="gl-flex gl-flex-col gl-items-center gl-justify-center gl-text-center"
      data-testid="dropzone-area"
    >
      <gl-animated-upload-icon-stub
        class="gl-mb-3"
      />
      <p
        class="gl-mb-0"
        data-testid="upload-text"
      >
        Drop or
        <gl-link-stub>
          upload
        </gl-link-stub>
        files to attach
      </p>
    </div>
  </button>
  <input
    accept="image/*"
    class="hide"
    multiple=""
    name="upload_file"
    type="file"
  />
  <transition-stub
    name="upload-dropzone-fade"
  >
    <div
      class="card gl-absolute gl-flex gl-h-full gl-items-center gl-justify-center gl-p-4 gl-w-full upload-dropzone-border upload-dropzone-overlay"
      style=""
    >
      <div
        class="gl-max-w-1/2 gl-text-center"
      >
        <h3>
          Oh no!
        </h3>
        <span>
          You are trying to upload something other than an image. Please upload a .png, .jpg, .jpeg, .gif, .bmp, .tiff or .ico.
        </span>
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`Upload dropzone component when dragging renders correct template when drag event is empty 1`] = `
<div
  class="gl-relative gl-w-full"
>
  <button
    class="card gl-h-full gl-items-center gl-justify-center gl-mb-0 gl-px-5 gl-py-4 gl-w-full upload-dropzone-border upload-dropzone-card"
    type="button"
  >
    <div
      class="gl-flex gl-flex-col gl-items-center gl-justify-center gl-text-center"
      data-testid="dropzone-area"
    >
      <gl-animated-upload-icon-stub
        class="gl-mb-3"
      />
      <p
        class="gl-mb-0"
        data-testid="upload-text"
      >
        Drop or
        <gl-link-stub>
          upload
        </gl-link-stub>
        files to attach
      </p>
    </div>
  </button>
  <input
    accept="image/*"
    class="hide"
    multiple=""
    name="upload_file"
    type="file"
  />
  <transition-stub
    name="upload-dropzone-fade"
  >
    <div
      class="card gl-absolute gl-flex gl-h-full gl-items-center gl-justify-center gl-p-4 gl-w-full upload-dropzone-border upload-dropzone-overlay"
      style=""
    >
      <div
        class="gl-max-w-1/2 gl-text-center"
      >
        <h3>
          Oh no!
        </h3>
        <span>
          You are trying to upload something other than an image. Please upload a .png, .jpg, .jpeg, .gif, .bmp, .tiff or .ico.
        </span>
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`Upload dropzone component when dragging renders correct template when dragging stops 1`] = `
<div
  class="gl-relative gl-w-full"
>
  <button
    class="card gl-h-full gl-items-center gl-justify-center gl-mb-0 gl-px-5 gl-py-4 gl-w-full upload-dropzone-border upload-dropzone-card"
    type="button"
  >
    <div
      class="gl-flex gl-flex-col gl-items-center gl-justify-center gl-text-center"
      data-testid="dropzone-area"
    >
      <gl-animated-upload-icon-stub
        class="gl-mb-3"
      />
      <p
        class="gl-mb-0"
        data-testid="upload-text"
      >
        Drop or
        <gl-link-stub>
          upload
        </gl-link-stub>
        files to attach
      </p>
    </div>
  </button>
  <input
    accept="image/*"
    class="hide"
    multiple=""
    name="upload_file"
    type="file"
  />
  <transition-stub
    name="upload-dropzone-fade"
  >
    <div
      class="card gl-absolute gl-flex gl-h-full gl-items-center gl-justify-center gl-p-4 gl-w-full upload-dropzone-border upload-dropzone-overlay"
      style="display: none;"
    >
      <div
        class="gl-max-w-1/2 gl-text-center"
      >
        <h3>
          Oh no!
        </h3>
        <span>
          You are trying to upload something other than an image. Please upload a .png, .jpg, .jpeg, .gif, .bmp, .tiff or .ico.
        </span>
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`Upload dropzone component when dragging with design upload overlay enabled renders component with requires classes when design upload overlay is true 1`] = `
<div
  class="gl-w-full"
>
  <button
    class="card gl-h-full gl-items-center gl-justify-center gl-mb-0 gl-px-5 gl-py-4 gl-w-full upload-dropzone-border upload-dropzone-card"
    type="button"
  >
    <div
      class="gl-flex gl-flex-col gl-items-center gl-justify-center gl-text-center"
      data-testid="dropzone-area"
    >
      <gl-animated-upload-icon-stub
        class="gl-mb-3"
      />
      <p
        class="gl-mb-0"
        data-testid="upload-text"
      >
        Drop or
        <gl-link-stub>
          upload
        </gl-link-stub>
        files to attach
      </p>
    </div>
  </button>
  <input
    accept="image/*"
    class="hide"
    multiple=""
    name="upload_file"
    type="file"
  />
  <transition-stub
    name="upload-dropzone-fade"
  >
    <div
      class="card design-upload-dropzone-overlay gl-absolute gl-border-1 gl-border-blue-500 gl-border-dashed gl-flex gl-h-full gl-items-center gl-justify-center gl-p-4 gl-w-full gl-z-200"
      style=""
    >
      <div
        class="gl-absolute gl-bg-blue-950 gl-bottom-6 gl-flex gl-items-center gl-px-3 gl-py-2 gl-rounded-base gl-text-white"
        data-testid="design-upload-overlay"
      >
        <gl-animated-upload-icon-stub
          ison="true"
          name="upload"
        />
        <span
          class="gl-ml-2"
        >
          Drop your images to start the upload.
        </span>
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`Upload dropzone component when no slot provided renders default dropzone card 1`] = `
<div
  class="gl-relative gl-w-full"
>
  <button
    class="card gl-h-full gl-items-center gl-justify-center gl-mb-0 gl-px-5 gl-py-4 gl-w-full upload-dropzone-border upload-dropzone-card"
    type="button"
  >
    <div
      class="gl-flex gl-flex-col gl-items-center gl-justify-center gl-text-center"
      data-testid="dropzone-area"
    >
      <gl-animated-upload-icon-stub
        class="gl-mb-3"
      />
      <p
        class="gl-mb-0"
        data-testid="upload-text"
      >
        Drop or
        <gl-link-stub>
          upload
        </gl-link-stub>
        files to attach
      </p>
    </div>
  </button>
  <input
    accept="image/*"
    class="hide"
    multiple=""
    name="upload_file"
    type="file"
  />
  <transition-stub
    name="upload-dropzone-fade"
  >
    <div
      class="card gl-absolute gl-flex gl-h-full gl-items-center gl-justify-center gl-p-4 gl-w-full upload-dropzone-border upload-dropzone-overlay"
      style="display: none;"
    >
      <div
        class="gl-max-w-1/2 gl-text-center"
      >
        <h3>
          Incoming!
        </h3>
        <span>
          Drop your files to start your upload.
        </span>
      </div>
    </div>
  </transition-stub>
</div>
`;

exports[`Upload dropzone component when slot provided renders dropzone with slot content 1`] = `
<div
  class="gl-relative gl-w-full"
>
  <div>
    dropzone slot
  </div>
  <transition-stub
    name="upload-dropzone-fade"
  >
    <div
      class="card gl-absolute gl-flex gl-h-full gl-items-center gl-justify-center gl-p-4 gl-w-full upload-dropzone-border upload-dropzone-overlay"
      style="display: none;"
    >
      <div
        class="gl-max-w-1/2 gl-text-center"
      >
        <h3>
          Incoming!
        </h3>
        <span>
          Drop your files to start your upload.
        </span>
      </div>
    </div>
  </transition-stub>
</div>
`;
